moxy.py

```diff
@@ -380,6 +380,84 @@ def replace_in_content(replace: Union[str,list,dict], content):
             ctx.log.error("Invalid JSON: {}: after replace: {}".format(error, replace))
     return content
 
+def lambda_with_args(lambda_func: str, index: int):
+    """
+    Given a string representation of a lambda function and the corresponding code
+    block delimiter index, returns `lambda_func` as a string representing a lambda
+    function, inserting an optional argument if necessary.
+    """
+    return re.sub("^\\s*lambda\\s*$", "lambda _", lambda_func[:index]) + lambda_func[index:]
+
+def replace_lambda_in_dict(replace_lambda: dict, dictionary):
+    """
+    For each value in `replace_lambda`, performs corresponding replacement (dict
+    update) in `dictionary`.
+    """
+    for key in replace_lambda:
+        replace_lambda_value = replace_lambda[key]
+        if isinstance(replace_lambda_value, dict):
+            dictionary[key] = replace_lambda_in_dict(replace_lambda_value, dictionary.get(key, {}))
+        else:
+            index = replace_lambda_value.find(":")
+            if index != -1:
+                replace_lambda_value = lambda_with_args(replace_lambda_value, index)
+                try:
+                    code_object = compile(replace_lambda_value, "<lambda>", "eval")
```

Comment:
I wonder if these compiled lambdas should be cached like compiled regexes are? Also, several lines of duplicated code between `replace_lambda_in_dict` and `replace_lambda_in_content`, maybe move to a function called by both?

Reply:
Great feedback, I will refactor accordingly.